src/x/errors/errors.go

```diff
@@ -130,6 +130,40 @@ func GetInnerInvalidParamsError(err error) error {
 	return nil
 }
 
+type resourceExhaustedError struct {
+	containedError
+}
+
+// NewResourceExhaustedError creates a new resource exhausted error.
+func NewResourceExhaustedError(inner error) error {
+	return resourceExhaustedError{containedError{inner}}
+}
+
+func (e resourceExhaustedError) Error() string {
+	return e.inner.Error()
+}
+
+func (e resourceExhaustedError) InnerError() error {
+	return e.inner
+}
+
+// IsResourceExhausted returns true if this is a resource exhausted error.
+func IsResourceExhausted(err error) bool {
+	return GetInnerResourceExhaustedError(err) != nil
+}
```

Comment:
NB: these methods might behave in a confusing way. E.g. `NewResourceExhaustedError(NewInvalidParamError(err))` could be treated as both depending on which of `IsResourceExhausted()` or `IsInvalidParam()` was called first.